src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo;
+  deleteHandler: ((todoId: number) => Promise<unknown>);
+  completedIsRemoving?: boolean;
+  toggleTodoStatus: (todoId: number, completed: boolean) => Promise<unknown>;
+  todosStatusIsChanging?: boolean;
+  toggleAllIsActive?: boolean;
+  renameTodo: (todoId: number, title: string) => Promise<unknown>;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteHandler,
+  completedIsRemoving,
+  toggleTodoStatus,
+  todosStatusIsChanging,
+  toggleAllIsActive,
+  renameTodo,
+}) => {
+  const { completed, title, id } = todo;
+
+  const [isDeleting, setIsDeleting] = useState(false);
+  const [isChanging, setIsChanging] = useState(false);
+  const [titleIsChanging, setTitleIsChanging] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const removeTodo = () => {
+    setIsDeleting(true);
+    deleteHandler(id).finally(() => setIsDeleting(false));
+  };
+
+  const changeStatus = () => {
+    setIsChanging(true);
+    toggleTodoStatus(id, !completed).finally(() => setIsChanging(false));
+  };
+
+  const changeTodoTitle = (newTodoTitle: string) => {
+    setIsChanging(true);
+    renameTodo(id, newTodoTitle).finally(() => setIsChanging(false));
+  };
+
+  const saveTitleChanges = () => {
+    setTitleIsChanging(false);
+    if (title === newTitle) {
+      return;
+    }
+
+    if (newTitle === '') {
```

Comment:
```suggestion
    if (!newTitle.trim()) {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim